### PR TITLE
Inject current branch slug into the documentation link

### LINKED ISSUE #8955: Documentation links are broken
## Current behaviour

Documentation pages are inaccessible from the app.

E.g. the Settings dialog:
image

Click on the blue ℹ in this dialog will try to open the following URLs:
* https://git-extensions-documentation.readthedocs.io/settings.html#git-extensions-appearance-author-images-avatar-provider
* https://git-extensions-documentation.readthedocs.io/settings.html#git-extensions-appearance-author-images-avatar-fallback
And a user is greeted with the following:
image

Also open `FormClone` dialog and click on [?] button:
image
It opens https://git-extensions-documentation.readthedocs.io/getting_started.html#clone-repository, which is also a 404.

## Expected behaviour

All links are correctly resolved and opened for the _latest_ version.

## Did this work in previous version of GitExtensions

Very likely a regression introduced in https://github.com/gitextensions/gitextensions/pull/8838 and https://github.com/gitextensions/gitextensions/pull/8893.

## Environment

- GitExtensions version: 12615869044a4db1f95609c05af93665045eb181

### PR DESCRIPTION
**Pull request by RussKie**
Resolves #8955

### CODE REVIEW

**Comment by gerhardol** [comment]
Better, but prefix need to be either /en/master or /en/release-xxx so logic (and XML doc) need to be adjusted.

**Comment by RussKie** [comment]
> Better, but prefix need to be either /en/master or /en/release-xxx so logic (and XML doc) need to be adjusted.

Why?

You said you fixed the redirection, so https://git-extensions-documentation.readthedocs.io/settings.html#git-extensions-appearance-author-images-avatar-provider now correctly resolves to https://git-extensions-documentation.readthedocs.io/en/release-3.4/settings.html#git-extensions-appearance-author-images-avatar-provider.
So the master branch points to the latest.

Meanwhile this change makes a release build point to the respective doc's branch.

Am I overlooking something?

**Comment by gerhardol** [comment]
> Am I overlooking something?

With the issues I saw with the redirection, I believe it is better to not rely on it.

If / points to the development version, that is what users will see by default too (currently 3.5 but it should be 3.6 or so).

**Comment by RussKie** [comment]
Ok, added "en/latest" for all non "release/x.y" branches

File: GitCommands/Settings/AppSettings.cs [hunk 1]
```diff
@@ -117,6 +119,41 @@ static AppSettings()
...
+        }
+
+        internal static void SetDocumentationBaseUrl(string currentGitBranch)
+        {
+            if (_documentationBaseUrl is not null)
+            {
+                throw new InvalidOperationException("Documentation base URL can only be set once");
+            }
+
+            string? docVersion = "en/latest/";
```
Comment (gerhardol) [inline GitCommands/Settings/AppSettings.cs:142]: (not null)
Suggest we use master to point out that this is dev version of the doc (where the version could be updated)
We have used 'latest' for the released version so far

```suggestion
            string docVersion = "en/master/";
```

Comment (RussKie) [reply GitCommands/Settings/AppSettings.cs:142]: Given the controversies around the "master", let's have it "en/dev"

Comment (gerhardol) [reply GitCommands/Settings/AppSettings.cs]: Then we chanhange the name of the "master" branch too.
I prefer "main" though, just as GH changed (and some longtime ClearCase usage).

**Comment by RussKie** [comment]
sure

**Review by gerhardol — APPROVED** [review]

**Comment by gerhardol** [comment]
I reordered Settings-Scripts so I could push main branch and build on read-the-docs so links in the app works.

**Pull request merged**